libraries/Microsoft.Bot.Builder.Azure/AzureBlobTranscriptStore.cs

```diff
@@ -0,0 +1,252 @@
+                        if (conversations.Count == pageSize)
+                            break;
+                    }
+                }
+
+                if (segment.ContinuationToken != null)
+                    token = segment.ContinuationToken;
+            } while (token != null && conversations.Count < pageSize);
+
+            var pagedResult = new PagedResult<Conversation>();
+            pagedResult.Items = conversations.ToArray();
+
+            if (pagedResult.Items.Length == 20)
+                pagedResult.ContinuationToken = pagedResult.Items.Last().Id;
+
+            return pagedResult;
+        }
+
+        public async Task DeleteConversation(string channelId, string conversationId)
+        {
+            if (channelId == null)
+                throw new ArgumentNullException($"{nameof(channelId)} should not be null");
+
+            if (conversationId == null)
+                throw new ArgumentNullException($"{nameof(conversationId)} should not be null");
+
+            var dirName = GetDirName(channelId, conversationId);
+            var dir = this.Container.Value.GetDirectoryReference(dirName);
+            BlobContinuationToken token = null;
+            List<CloudBlockBlob> blobs = new List<CloudBlockBlob>();
+            do
+            {
+                var segment = await dir.ListBlobsSegmentedAsync(false, BlobListingDetails.None, null, token, null, null);
+                foreach (var blob in segment.Results.Cast<CloudBlockBlob>())
+                {
+                    await blob.DeleteIfExistsAsync();
+                }
+                if (segment.ContinuationToken != null)
+                    token = segment.ContinuationToken;
+            } while (token != null);
+        }
+
+        /// <summary>
+        /// Get a blob name validated representation of an entity
+        /// </summary>
+        /// <param name="key">The key used to identify the entity</param>
+        /// <returns></returns>
+        private string GetBlobName(IActivity activity)
+        {
+            var blobName = $"{SanitizeKey(activity.ChannelId)}/{SanitizeKey(activity.Conversation.Id)}/{activity.Timestamp.Value.Ticks.ToString("x")}-{SanitizeKey(activity.Id)}.json";
+            NameValidator.ValidateBlobName(blobName);
+            return blobName;
+        }
+
+        private string GetDirName(string channelId, string conversationId = null)
+        {
+            string dirName = "";
+            if (conversationId != null)
+                dirName = $"{SanitizeKey(channelId)}/{SanitizeKey(conversationId)}";
+            else
+                dirName = $"{SanitizeKey(channelId)}";
+            // NameValidator.ValidateDirectoryName(dirName);
+            return dirName;
+        }
+
+        private static Lazy<Dictionary<char, string>> badChars = new Lazy<Dictionary<char, string>>(() =>
+        {
+            char[] badChars = new char[] { '\\', '?', '/', '#', '\t', '\n', '\r' };
```

Comment:
Where does this list of bad characters come from? How do I know if it's correct? 

For example, this set of docs indicates that the characters listed here are not the right ones:
https://blogs.msdn.microsoft.com/jmstall/2014/06/12/azure-storage-naming-rules/